### PR TITLE
Add clippy::indexing_slicing to the list of ignored lints

### PR DESCRIPTION
**Pull request by cfcosta**
I am currently using `valuable` on a project that has most clippy lints enabled, and it is showing warnings because it uses direct array indexing to access the fields on the generated macro code. Unfortunately, there's no way to ignore it directly, as it gets "imported" to application code on macro expansion.

This PR adds `clippy::indexing_slicing` to the listing of ignored lints.

### CODE REVIEW

**Review by taiki-e — APPROVED** [review]
Thanks.

It is usually a bug that clippy warns of code generated by external macros, and it also seems to be a bug that this lint is triggered for static slices and immediate indexes whose exact length is known at compile time, so I recommend you report the bug to clippy.

That said, it is very easy to work around this by ignoring it on our part, so adding a allow attribute is a reasonable action.

**Comment by cfcosta** [comment]
@taiki-e will do. Does anyone else needs approving this for it to be merged? The test failures seem completely unrelated.

**Comment by taiki-e** [comment]
CI failure will be fixed in #119. I will merge this once that PR is merged.

**Pull request merged**
